src/main/java/com/cqse/teamscaleupload/OkHttpClientUtils.java

```diff
@@ -1,31 +1,118 @@
 package com.cqse.teamscaleupload;
 
+import java.io.FileInputStream;
+import java.io.IOException;
 import java.security.GeneralSecurityException;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
 import java.security.SecureRandom;
+import java.security.cert.CertificateException;
 import java.security.cert.X509Certificate;
 import java.util.concurrent.TimeUnit;
 
 import javax.net.ssl.SSLContext;
 import javax.net.ssl.SSLSocketFactory;
 import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
 import javax.net.ssl.X509TrustManager;
 
 import okhttp3.OkHttpClient;
 
+/**
+ * Utilities for creating an {@link OkHttpClient}
+ */
 public class OkHttpClientUtils {
 
-    public static OkHttpClient createClient(boolean validateSsl) {
+    /**
+     * Creates the {@link OkHttpClient} based on the given connection settings.
+     *
+     * @param trustStorePath     May be null if no trust store should be used.
+     * @param trustStorePassword May be null if no trust store should be used.
+     */
+    public static OkHttpClient createClient(boolean validateSsl, String trustStorePath, String trustStorePassword) {
         OkHttpClient.Builder builder = new OkHttpClient.Builder();
 
         setSensibleTimeouts(builder);
         builder.followRedirects(false).followSslRedirects(false);
 
+        if (trustStorePath != null) {
+            configureTrustStore(builder, trustStorePath, trustStorePassword);
+        }
         if (!validateSsl) {
             disableSslValidation(builder);
         }
+
         return builder.build();
     }
 
+    /**
+     * Reads the keystore at the given path and configures the builder so the {@link OkHttpClient}
+     * will accept the certificates stored in the keystore.
+     */
+    private static void configureTrustStore(OkHttpClient.Builder builder, String trustStorePath, String trustStorePassword) {
+        KeyStore keyStore = readKeyStore(trustStorePath, trustStorePassword);
+        try {
+            SSLContext sslContext = SSLContext.getInstance("SSL");
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            trustManagerFactory.init(keyStore);
+            sslContext.init(null, trustManagerFactory.getTrustManagers(), new SecureRandom());
+
+            if (trustManagerFactory.getTrustManagers().length == 0) {
+                TeamscaleUpload.fail("No trust managers found. This is a bug. Please report it to CQSE.");
+            }
+
+            try {
```

Comment:
Can we avoid the nested try-catch here? Maybe extract the inner try-catch to a separate method?